### PR TITLE
[FLINK-1266] Generalize DistributedFileSystem implementation

### PR DESCRIPTION
**Pull request by rmetzger**
to HadoopFileSystem wrapper, which supports all subclasses of `org.apache.hadoop.fs.FileSystem`.
This allows us to let users use all file systems with support for HDFS.

The change has been tested with Tachyon, Google Cloud Storage Hadoop Adapter and HDFS. (I also contributed again to Tachyon)

The change also cleans up the Hadoop dependency exclusions.
I also found a nice little "bug" in the Instance class.

### CODE REVIEW

File: flink-addons/flink-tachyon/pom.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			<groupId>org.tachyonproject</groupId>
+			<artifactId>tachyon</artifactId>
+			<version>0.5.0</version>
+			<type>test-jar</type>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.jetty</groupId>
+			<artifactId>jetty-util</artifactId>
+			<version>7.6.8.v20121106</version>
```
Comment (StephanEwen) [inline flink-addons/flink-tachyon/pom.xml]: This may conflict with the jetty version used by flink-runtime for the web info server

**Comment by StephanEwen** [comment]
This pull request moves the hadoop transitive dependency exclusions to the dependency management. Have you checked that this works properly? I ran into an issue there a while back where it did not properly work. Cannot quite recall why, though.

File: flink-addons/flink-tachyon/pom.xml (hunk 1 above)
Comment (rmetzger) [inline flink-addons/flink-tachyon/pom.xml]: I will double check what maven makes out of these two versions.
But the Jetty version is `test` scoped and should not influence the regular build.

it also seems that I forgot to scope the regular tachyon dependency to `test`.

**Comment by rmetzger** [comment]
Yes, I found it very inconvenient to copy paste the exclusions everywhere. I will check if the exclusions still work correctly.

**Comment by rmetzger** [comment]
Update: I fixed a major bug with the "2.0.0-alpha" version on the `master`.
Currently, the master does not do any dependency exclusions for hadoop. This causes stuff like two asm versions in the lib path.

I also verified that the dependency exclusions in the `dependencyManagement` section are working. (Its also in the official maven documentation .. so it should work).

**Comment by StephanEwen** [comment]
Thanks, that looks good.

**Pull request closed without merging**